File: core/src/com/xenoage/zong/core/music/direction/Dynamic.java
package com.xenoage.zong.core.music.direction;

import com.xenoage.zong.core.music.MusicElementType;
import com.xenoage.zong.core.position.MP;

import lombok.Data;
import lombok.EqualsAndHashCode;


/**
 * Class for a dynamic sign like forte, piano, sforzando and so on.
 *
 * @author Andreas Wenger
 */
@Data @EqualsAndHashCode(callSuper=false)
public final class Dynamic
	extends Direction {

	private final DynamicValue value;

	@Override public MusicElementType getMusicElementType() {
		return MusicElementType.Dynamic;
	}
	
	@Override public MP getMP() {
		return MP.getMPFromParent(this);
	}
	
}

File: core/src/com/xenoage/zong/core/music/direction/Dynamic.kt
package com.xenoage.zong.core.music.direction

import com.xenoage.zong.core.music.chord.Chord
import com.xenoage.zong.core.music.format.Positioning


/**
 * Class for a dynamic sign like forte, piano, sforzando and so on.
 */
class Dynamic(
		val value: DynamicValue
) : Direction {

	override var positioning: Positioning? = null

	override var parent: Chord? = null

}
